tests/config/cliEnv.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Env, TestInfo, WorkerInfo } from '../folio/out';
+import { PageEnv } from './browserEnv';
+import { CLIMock, CLITestArgs, Recorder } from './cliTest';
+import * as http from 'http';
+import { recorderPageGetter } from '../../test/utils';
+
+export class CLIEnv extends PageEnv implements Env<CLITestArgs> {
+  private _server: http.Server | undefined;
+  private _handler = (req: http.IncomingMessage, res: http.ServerResponse) => res.end();
+  private _port: number;
+  private _cli: CLIMock | undefined;
+
+  async beforeAll(workerInfo: WorkerInfo) {
+    await super.beforeAll(workerInfo);
+
+    this._port = 9907 + workerInfo.workerIndex;
+    this._server = http.createServer((req: http.IncomingMessage, res: http.ServerResponse) => this._handler(req, res)).listen(this._port);
+  }
+
+  private _runCLI(args: string[]) {
+    this._cli = new CLIMock(this._browserName, this._browserOptions.channel, !!this._browserOptions.headless, args);
+    return this._cli;
+  }
+
+  async beforeEach(testInfo: TestInfo) {
+    const result = await super.beforeEach(testInfo);
+    const { page, context, toImpl } = result;
+    return {
+      ...result,
+      httpServer: {
+        setHandler: newHandler => this._handler = newHandler,
+        PREFIX: `http://127.0.0.1:${this._port}`,
+      },
+      runCLI: this._runCLI.bind(this),
+      openRecorder: async () => {
```

Comment:
Why does this change from `recorder` to `openRecorder`. It's because the new system doesn't magically resolve the fixtures I use?

Reply:
Exactly. We use cliTest for both recorder tests and runCLI tests, and I didn't want to open recorder for the latter.

Reply:
Is there still magic for the built in fixtures? Or do we launch a browser per test now?